### PR TITLE
[ESPN] Extract duration and update tests

### PR DESCRIPTION
**Pull request by ischmidt20**
<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### Description of your *pull request* and other information

</details>

Extracts duration from ESPN+ / WatchESPN videos and updates the tests to reflect currently available videos.

### CODE REVIEW

File: yt_dlp/extractor/espn.py [hunk 1]
```diff
@@ -283,34 +284,37 @@ def _real_extract(self, url):
...
+            'title': 'Dynamo Dresden vs. VfB Stuttgart (Round #1) (German Cup)',
+            'duration': 8335,
             'thumbnail': 'https://s.secure.espncdn.com/stitcher/artwork/collections/media/bd1f3d12-0654-47d9-852e-71b85ea695c7/16x9.jpg?timestamp=202201112217&showBadge=true&cb=12&package=ESPN_PLUS',
         },
         'params': {
             'skip_download': True,
         },
     }, {
-        'url': 'https://www.espn.com/espnplus/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421',
+        'url': 'https://www.espn.com/watch/player/_/id/47950544-b39f-4e47-a982-b2132d20c8d1',
```
Comment (pukkandan) [inline yt_dlp/extractor/espn.py:311]: this is supposed to test `espnplus/` URL

File: yt_dlp/extractor/espn.py [hunk 2]
```diff
@@ -328,9 +332,12 @@ def _call_bamgrid_api(self, path, video_id, payload=None, headers={}):
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
-        video_data = self._download_json(
+        cdn_data = self._download_json(
             f'https://watch-cdn.product.api.espn.com/api/product/v3/watchespn/web/playback/event?id={video_id}',
-            video_id)['playbackState']
+            video_id)
+
+        duration = traverse_obj(cdn_data, ('tracking', 'duration'))
```
Comment (pukkandan) [inline yt_dlp/extractor/espn.py:339]: can be inlined in the return

**Review by pukkandan — CHANGES_REQUESTED** [review]
everything else looks good

File: yt_dlp/extractor/espn.py (hunk 1 above)
Comment (ischmidt20) [reply yt_dlp/extractor/espn.py]: My mistake, reverted

**Pull request merged**